### PR TITLE
Add support for otlp trace exporters

### PR DESCRIPTION
**Pull request by Sovietaced**
## Tracking issue
Closes https://github.com/flyteorg/flyte/issues/5476

## Why are the changes needed?
Distributed traces can only be exported using Jaeger with the current code. The[ jaeger protocol has been deprecated in favor of the open telemetry protocol (OTLP)](https://opentelemetry.io/blog/2022/jaeger-native-otlp/). Additionally, if we're using OTEL client libraries Flyte should also just support their native protocol as well. 

We should also deprecate use of the Jaeger exporter and slate it for removal. 

## What changes were proposed in this pull request?

This pull request adds two new trace exporter strategies: `otlpGrpc` (OTLP over gRPC) and `otlpHttp` (OTLP over HTTP). 

## How was this patch tested?

This patch was tested locally in our non production environment.

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

File: flytestdlib/otelutils/config.go [hunk 1]
```diff
@@ -33,7 +35,7 @@ var (
 )
 
 type Config struct {
-	ExporterType ExporterType `json:"type" pflag:",Sets the type of exporter to configure [noop/file/jaeger]."`
+	ExporterType ExporterType `json:"type" pflag:",Sets the type of exporter to configure [noop/file/jaeger/otlpGrpc/otlpHttp]."`
 	FileConfig   FileConfig   `json:"file" pflag:",Configuration for exporting telemetry traces to a file"`
 	JaegerConfig JaegerConfig `json:"jaeger" pflag:",Configuration for exporting telemetry traces to a jaeger"`
```
Comment (Sovietaced) [inline flytestdlib/otelutils/config.go:40]: All the OTEL exporter configuration can come from environment variables so I didn't really see any value in 1) deviating from OTEL and 2) trying to copy all of the settings that can be configured via the OTEL environment variables.

**Comment by EngHabu** [comment]
@hamersaw @andrewwdye can you plz review this PR?

File: flytestdlib/otelutils/config.go (hunk 1 above)
Comment (andrewwdye) [reply flytestdlib/otelutils/config.go:40]: Good callout @Sovietaced. My hesitation is that it deviates with other trace configuration and how flyte components are configured generally. It adds ambiguity to a component's behavior and how to configure it. I'm inclined to keep consistent.

If it's helpful I have a private version of this to use OTEL + expose some sampling config. I could pull that in instead and save you some cycles. Wdyt?

Comment (Sovietaced) [reply flytestdlib/otelutils/config.go:40]: I can see the user experience being a little nicer for some folks if more configuration came through the YAML dictionaries in the helm chart but I'm personally not interested in trying to duplicate the OTEL standardized environment variables so you're probably better of pulling in changes from your fork.

Comment (andrewwdye) [reply flytestdlib/otelutils/config.go:40]: Sounds good. I have a draft up at https://github.com/flyteorg/flyte/pull/5504. Let me pull in a few more of the changes you proposed here as well

**Comment by Sovietaced** [comment]
Closing in favor of: https://github.com/flyteorg/flyte/pull/5504

**Pull request closed without merging**